packages/core/src/model.js

```diff
@@ -1690,8 +1690,10 @@ ${associationOwner._getAssociationDebugList()}`);
     // use a subquery to get the count
     if (options.group && options.countGroupedRows) {
       const query = removeTrailingSemicolon(this.queryGenerator.selectQuery(this.table, options));
+      const dialect = this.sequelize.dialect.name;
 
-      const queryCountAll = `Select COUNT(*) AS count FROM (${query}) AS Z`;
+      // Oracle doesn't support 'AS' keyword for aliasing tables
+      const queryCountAll = `Select COUNT(*) AS count FROM (${query}) ${dialect !== 'oracle' ? 'AS' : ''} Z`;
```

Comment:
I think we should move this to the queryGenerator. That also allows us to use the `getAliasToken` function